packages/document/docs/en/api/javascript-api/instance.mdx

```diff
@@ -267,6 +267,124 @@ await rsbuild.startDevServer({
 });
 ```
 
+## rsbuild.createDevServer
+
+Rsbuild comes with a built-in dev server designed to improve the development experience. When you run the `rsbuild dev` command, the server will start, providing features such as page preview, routing, and hot module reloading.
+
+If you want to integrate Rsbuild DevServer into an existing Server, you can use this method to obtain the instance methods of Rsbuild DevServer and call them on demand.
+
+- **type:**
+
+```ts
+type RsbuildDevServer = {
+  /** start the Rsbuild DevServer */
+  listen: () => Promise<{
+    urls: string[];
+    port: number;
+    server: Server;
+  }>;
+
+  /** The following APIs will be used when you use a custom server */
+
+  /** The resolved rsbuild server config */
+  config: {
+    devServerConfig: DevConfig & ServerConfig;
+    port: number;
+    host: string;
+    https: boolean;
+    defaultRoutes: Routes;
+  };
+
+  /**
+   * connect app instance.
+   *
+   * Can be used to attach custom middlewares to the dev server.
+   */
+  middlewares: Middlewares;
+
+  /** Notify Rsbuild that the custom server has started */
+  afterListen: (options?: { port?: number; routes?: Routes }) => Promise<void>;
```

Comment:
Why the `afterListen` needs to provide the `port` and `routes` params?

Reply:
Deleted. This value is required in `onAfterStartDevServer` hook and will be used in startUrl and printUrls, but for users, they don't need to customize them, they can directly configure `dev.startUrl` and `server.printUrls`.